lib/Data/Chronicle/Reader.pm

```diff
@@ -205,6 +205,35 @@ sub get_for_period {
     return \@result;
 }
 
+=head3 C<< my $data = get_history("category1", "name1", 1) >>
+
+Query Pg archive for the data under "category1::name1" at the provided number of revisions in the past.
+
+=cut
+
+sub get_history {
+    my $self     = shift;
+    my $category = shift;
+    my $name     = shift;
+    my $rev      = shift;
+
+    die "Revision must be >= 0" if (!defined $rev || $rev < 0);
+    die "Requesting for historical data without a valid DB connection [$category,$name,$rev]" if not defined $self->dbic;
+
+    my $db_data = $self->dbic->run(
+        fixup => sub {
+            $_->selectall_hashref(q{SELECT * FROM chronicle where category=? and name=? order by timestamp desc limit ?},
```

Comment:
`$_->selectrow_array(q{select value from chronicle where category=? and name=? order by timestamp desc limit 1 offset ?}, undef, $category, $name, $rev);` will be better.